ams/server/src/main/java/com/netease/arctic/server/table/TableSelfOptimizingMetrics.java

```diff
@@ -0,0 +1,350 @@
+      defineGauge("table_optimizing_status_committing_duration_mills")
+          .withDescription("Duration in seconds after table be in committing state")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  // table optimizing process count metrics
+  public static final MetricDefine TABLE_OPTIMIZING_PROCESS_TOTAL_COUNT =
+      defineCounter("table_optimizing_process_total_count")
+          .withDescription("Count of all process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_PROCESS_FAILED_COUNT =
+      defineCounter("table_optimizing_process_failed_count")
+          .withDescription("Count of failed process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_MINOR_TOTAL_COUNT =
+      defineCounter("table_optimizing_minor_total_count")
+          .withDescription("Count of minor process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_MINOR_FAILED_COUNT =
+      defineCounter("table_optimizing_minor_failed_count")
+          .withDescription("Count of failed minor process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_MAJOR_TOTAL_COUNT =
+      defineCounter("table_optimizing_major_total_count")
+          .withDescription("Count of major process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_MAJOR_FAILED_COUNT =
+      defineCounter("table_optimizing_major_failed_count")
+          .withDescription("Count of failed major process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_FULL_TOTAL_COUNT =
+      defineCounter("table_optimizing_full_total_count")
+          .withDescription("Count of full process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_FULL_FAILED_COUNT =
+      defineCounter("table_optimizing_full_failed_count")
+          .withDescription("Count of failed full process since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  // table optimizing process status metrics
+  public static final MetricDefine TABLE_OPTIMIZING_STATUS_IN_IDLE =
+      defineGauge("table_optimizing_status_in_idle")
+          .withDescription("If currently table is in status idle")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_STATUS_IN_PENDING =
+      defineGauge("table_optimizing_status_in_pending")
```

Comment:
optimizing status is quite a new concept(or conflicts with OptimizingStatus)

why not make them all unified?